cmd/minishift/cmd/profile/profile_set.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package profile
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/docker/machine/libmachine"
+	"github.com/minishift/minishift/pkg/minikube/constants"
+	minishiftConfig "github.com/minishift/minishift/pkg/minishift/config"
+	"github.com/minishift/minishift/pkg/minishift/oc"
+	profileActions "github.com/minishift/minishift/pkg/minishift/profile"
+	"github.com/minishift/minishift/pkg/util/os/atexit"
+	"github.com/spf13/cobra"
+)
+
+const (
+	emptyProfileError  = "You must provide the profile name. Run `minishift profile list` to view profiles"
+	extraArgumentError = "You have provided more arguments than required. You must provide a single profile name"
+)
+
+var profileSetCmd = &cobra.Command{
+	Use:   "set",
+	Short: "Sets the default profile for Minishift",
+	Long:  "Sets the default profile for Minishift. After you set the profile, all commands will use the profile by default",
+	Run: func(cmd *cobra.Command, args []string) {
+		var doesProfileExist = false
+		if len(args) == 0 {
+			atexit.ExitWithMessage(1, emptyProfileError)
+		} else if len(args) > 1 {
+			atexit.ExitWithMessage(1, extraArgumentError)
+		}
+
+		profileName := args[0]
+
+		//check if the profile is present in the AllInstancesConfig
+		profileList := profileActions.GetProfileNameList()
```

Comment:
I am wondering whether we really should store the list of profiles in the config file. Why not just store the active profile as a single entry. Then when it comes to determining the list of profiles we look at the actual file system. This would prevent that the config files gets out of sync with the actual file system.

Reply:
agree. removing a VM by hand will not be picked up correctly in the current form.

Reply:
or moving one from an external disk